main.c

```diff
@@ -585,31 +581,86 @@ init_kms(struct vkcube *vc)
    vc->image_format = VK_FORMAT_R8G8B8A8_SRGB;
    init_vk_objects(vc);
 
-   PFN_vkCreateDmaBufImageINTEL create_dma_buf_image =
-      (PFN_vkCreateDmaBufImageINTEL)vkGetDeviceProcAddr(vc->device, "vkCreateDmaBufImageINTEL");
-
    for (uint32_t i = 0; i < 2; i++) {
       struct vkcube_buffer *b = &vc->buffers[i];
       int fd, stride, ret;
+     
+      uint64_t mod = DRM_FORMAT_MOD_LINEAR;
 
-      b->gbm_bo = gbm_bo_create(vc->gbm_device, vc->width, vc->height,
-                                GBM_FORMAT_XRGB8888, GBM_BO_USE_SCANOUT);
+      uint32_t flags = 0;
+      flags |=  GBM_BO_USE_SCANOUT |
+	      (vc->protected? GBM_BO_USE_PROTECTED : 0);
+      b->gbm_bo = gbm_bo_create_with_modifiers2(vc->gbm_device, vc->width, vc->height, GBM_FORMAT_XRGB8888, &mod, 1, flags);
```

Comment:
Code style. This codebase tries to wrap at 80 columns. Please wrap the long line.